src/pages/Integrations/SplunkSetup/useSplunkSetup.tsx

```diff
@@ -0,0 +1,207 @@
+import { useUpdateBehaviorGroupActionsMutation } from '../../../services/Notifications/UpdateBehaviorGroupActions';
+import { useGetAllEventTypes } from '../../../services/GetEventTypes';
+import { useGetAnyBehaviorGroupByNotification } from '../../../services/Notifications/GetBehaviorGroupByNotificationId';
+import { linkBehaviorGroupAction } from '../../../services/Notifications/LinkBehaviorGroup';
+
+export const SPLUNK_GROUP_NAME = 'SPLUNK_INTEGRATION';
+export const SPLUNK_INTEGRATION_NAME = 'SPLUNK_AUTOMATION';
+export const SPLUNK_BEHAVIOR_GROUP_NAME = 'SPLUNK_AUTOMATION_GROUP';
+export const BUNDLE_NAME = 'rhel';
+
+interface SplunkEventsDef {
+    [Identifier: string]: string | string[]
+}
+
+const DEFAULT_SPLUNK_EVENTS : SplunkEventsDef = {
+    advisor: '*',
+    policies: '*',
+    drift: '*',
+};
+
+export const useSplunkSetup = () => {
+    const createSplunkIntegration = useCreateSplunkIntegration();
+    const createSplunkBehaviorGroup = useCreateSplunkBehaviorGroup();
+    const updateSplunkBehaviorActions = useUpdateSplunkBehaviorActions();
+    const attachEvents = useAttachEventsToSplunk();
+
+    return async ({ hecToken, splunkServerHostName }, onProgress) => {
+        const integrationName = SPLUNK_INTEGRATION_NAME;
+        const behaviorGroupName = SPLUNK_BEHAVIOR_GROUP_NAME;
+        const bundleName = BUNDLE_NAME;
+        const events = DEFAULT_SPLUNK_EVENTS;
+
+        onProgress('Creating Splunk Integration...');
+        const integration = await createSplunkIntegration({ integrationName, hecToken, splunkServerHostName });
+        onProgress(`  ${integrationName} integration created.`);
+
+        onProgress('Creating Splunk Behavior Group...');
+        const behaviorGroup = await createSplunkBehaviorGroup({ behaviorGroupName, bundleName });
+        onProgress(`  ${behaviorGroupName} behavior group created.`);
+
+        onProgress('Adding Splunk integration as an action behavior group...');
+        await updateSplunkBehaviorActions(behaviorGroup, integration);
+        onProgress('  Added.');
+
+        onProgress('Adding evets to the behavior group:');
+        await attachEvents(behaviorGroup, events, onProgress);
+    }
+};
+
+const useCreateSplunkIntegration = () => {
+    const { mutate } = useSaveIntegrationMutation();
+    return async ({ integrationName, splunkServerHostName, hecToken }) : Promise<Integration | undefined> => {
+        const newIntegration : NewIntegrationTemplate<IntegrationCamel> = {
+            type: IntegrationType.SPLUNK,
+            name: integrationName,
+            url: splunkServerHostName,
+            secretToken: hecToken,
+            isEnabled: true,
+            sslVerificationEnabled: true,
+        };
+
+        const { errorObject, payload } = await mutate(newIntegration);
+        if (errorObject) {
+            throw errorObject;
+        }
+
+        return payload?.value as Integration;
```

Comment:
The payload implements discrimination by `payload?.status` or `payload?.type`.
While is not required to modify this PR, one should be able to do:
```suggestion
        const { payload } = await mutate(newIntegration);
        if (payload?.status !== 200) {
            throw new Error('Invalid response');
        }

        return payload.value;
```

Reply:
I'm trusting the library to produce `errorObject` if there is an error. Comparing a single HTTP status codes is not ideal either, as it might be any of the `2XX` responses. In this POST case it should be `201 Created`. Check https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses.

I acknowledge that even the check of errorObject does not cover all cases, i.e. the case when the payload is not correct or passed.

Reply:
Also check https://marcin-piela.github.io/react-fetching-library/#/?id=error-boundaries

Reply:
Okay, I've added additional checks for `error`. Those would cover it, when the error object is not thrown. `error` is set for non `2xx` responses.

https://github.com/marcin-piela/react-fetching-library/blob/9b0abd81cf7494931ca144a1d64952f6e0d77b18/src/client/client.ts#L77
https://developer.mozilla.org/en-US/docs/Web/API/Response/ok

Reply:
One of these layers that were added is a validation layer - all the requests are validated (runtime) and discriminated types are created (compile time) to ensure the types are correct. 

It's maybe not ideal, but this allows us to detect api changes at compile time (given the openapi is updated).

Aside the `status` a type is added to the payload, in this case, if the type is `"Integration"` the payload.value is constrained to `Integration` and below cast is not needed anymore.

The implementation of these mechanisms needs to be documented and needs polishing to make it usable and friendly to the implementer, but it's something that is already there. 

![Screenshot_2022-04-13_06-50-25](https://user-images.githubusercontent.com/3845764/163184119-da687ac9-a70b-48f5-9de2-8d4f2053b42f.png)

An example on other request, the IDE will even help with hinting the possible types:

![Screenshot_2022-04-13_06-57-10](https://user-images.githubusercontent.com/3845764/163185281-f8fcf23e-259a-43d3-b854-0af2a6babf38.png)

![Screenshot_2022-04-13_06-58-51](https://user-images.githubusercontent.com/3845764/163185381-7c0b1641-361c-4fe7-a733-f291bd602edc.png)

I'm not requiring these changes on this PR. Current checks are OK  and can be refined later.

Reply:
Yeah, thanks. It would be great to have this generalized and avoid repeating it for each new api hook.

I've added couple of these notes to https://issues.redhat.com/browse/NOTIF-602